cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -870,12 +871,29 @@ def make_resource(self, data, **kwargs):
         return AmiSearchFilters(**data)
 
 
+class TimeoutsSchema(BaseSchema):
+    """Represent the schema of the Timeouts section."""
+
+    head_node_bootstrap_timeout = fields.Int(
+        validate=validate.Range(min=1), metadata={"update_policy": UpdatePolicy.UNSUPPORTED}
+    )
+    compute_node_bootstrap_timeout = fields.Int(
+        validate=validate.Range(min=1), metadata={"update_policy": UpdatePolicy.SUPPORTED}
+    )
+
+    @post_load()
+    def make_resource(self, data, **kwargs):
+        """Generate resource."""
+        return Timeouts(**data)
+
+
 class ClusterDevSettingsSchema(BaseDevSettingsSchema):
     """Represent the schema of Dev Setting."""
 
     cluster_template = fields.Str(metadata={"update_policy": UpdatePolicy.SUPPORTED})
     ami_search_filters = fields.Nested(AmiSearchFiltersSchema, metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
     instance_types_data = fields.Str(metadata={"update_policy": UpdatePolicy.SUPPORTED})
+    timeouts = fields.Nested(TimeoutsSchema, metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
```

Comment:
Does this imply that compute_node_bootstrap_timeout cannot be added later if Timeouts section is added later, even if UpdatePolicy is SUPPORTED for compute_node_bootstrap_timeout?

Reply:
Yes, this imply that compute_node_bootstrap_timeout cannot be added later if Timeouts section is added later, even if UpdatePolicy is SUPPORTED for compute_node_bootstrap_timeout
But making the policy SUPPORTED will allow adding and removing the whole section, if the headnode timeouts is changed, I am afraid that it will trigger headnode replacement.

Reply:
After some tests, the Update of headnode timeout only change the launch template, not triggering head node replacement, change Timeouts section to support update.